src/mixins/createCachedRenderMixin.ts

```diff
@@ -121,9 +126,15 @@ const createCachedRenderMixin: CachedRenderFactory = createStateful
 					props[key] = cachedRender.listeners[key];
 				}
 				const classes: { [index: string]: boolean; } = {};
+				const widgetClasses: string[] = widgetClassesMap.get(cachedRender) || [];
+
+				widgetClasses.forEach((c) => classes[c] = false);
+
 				if (cachedRender.classes) {
 					cachedRender.classes.forEach((c) => classes[c] = true);
 				}
+
+				widgetClassesMap.set(cachedRender, cachedRender.classes || []);
```

Comment:
The above `|| []` is for the `widgetClasses` not the `cachedRender.classes` which is externally defined in the widget's state (could be undefined), this could however be moved into the `if (cachedRender.classes)` above.